code/modules/clothing/clothing.dm

```diff
@@ -217,6 +217,41 @@
 		for(var/A in armor)
 			armor[A] -= rand(armor[A]/3, armor[A])
 
+// Redefines the attack proc so that when we attack somebody on fire on help intent with a piece of clothing, we have a small chance of putting out the fire.
+/obj/item/clothing/attack(mob/living/M as mob, mob/living/user as mob, def_zone, var/originator = null)
+	if(iscarbon(user)) // First, we check if it's a a carbon.
+		if(user.a_intent == I_HELP) // Then we check if he wants to help.
+			if(istype(src, /obj/item/clothing/under) || (istype(src, /obj/item/clothing/suit) && !istype(src, /obj/item/clothing/suit/space))) // Then we check if the item is either a jumpsuit or suit-but-not-a-spacesut
+				if(ishuman(M) && M.on_fire) // If he is on fire
+					/var/mob/living/carbon/human/target = new/mob/living/carbon/human()
+					target = M
+					// Good old polymorphism. We are allowed to do that because we checked if M was a human before.
+					// This allows us to check for their species
+					if(isplasmaman(target)) // Cannot put out plasmamen, else they could just go arround with a jumpsuit and not need a space suit.
+						for(var/mob/spectator in view())
+							to_chat(spectator, "<span class='warning'>\the [user] attempts to put out the fire on \the [target], but plasmafires are too hot. It is no use.</span>")
+						return
+					else
+						for(var/mob/spectator in view())
+							to_chat(spectator, "<span class='warning'>\the [user] attempts to put out the fire on \the [target].</span>")
+						if(prob(15))
+							M.ExtinguishMob()
+							for(var/mob/spectator in view())
+								to_chat(spectator, "<span class='notice'>\the [user] have put out the fire on \the [target].</span>")
+						return
+				else
+					..()
```

Comment:
oh my

Reply:
Yes it is ugly. How could I do it otherwise ?

Reply:
As long as you `return` inside the first part of your code you don't need any `else` at all, just `return ..()`